src/VisualStudio/TestUtilities/InProcess/SolutionExplorer_InProc.cs

```diff
@@ -122,7 +138,36 @@ public void AddProject(string projectName, string projectTemplate, string langua
         // TODO: Adjust language name based on whether we are using a web template
         private string GetProjectTemplatePath(string projectTemplate, string languageName)
         {
-            return _solution.GetProjectTemplate(_projectTemplates[projectTemplate], languageName);
+            var projectTemplateName = string.Empty;
+
+            switch (languageName.ToLower())
+            {
+                case "csharp":
+                {
+                    if (_csharpProjectTemplates.TryGetValue(projectTemplate, out projectTemplateName))
+                    {
+                        break;
+                    }
+                    goto default;
```

Comment:
Oh goodness, let's not use this...that's really hard to read. Can we just make this a few if checks? It's half the vertical space and a lot less surprising.